enhancements/monitoring/multi-tenant-alerting.md

```diff
@@ -0,0 +1,636 @@
+`enableUserAlertmanagerConfig` is missing, the default value is false.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |-
+    enableUserWorkload: true
+    alertmanagerMain:
+      enableUserAlertmanagerConfig: true
+```
+
+When `enableUserAlertmanagerConfig` is true, the cluster monitoring operator
+configures the Platform Alertmanager to reconcile `AlertmanagerConfig`
+resources from user namespaces as follows.
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: Alertmanager
+metadata:
+  name: main
+  namespace: openshift-monitoring
+spec:
+  alertmanagerConfigSelector: {}
+  alertmanagerConfigNamespaceSelector:
+    matchExpressions:
+    - key: openshift.io/cluster-monitoring
+      operator: NotIn
+      values:
+      - "true"
+    - key: openshift.io/user-monitoring
+      operator: NotIn
+      values:
+      - "false"
+  ...
+```
+
+To be consistent with what exists already for service/pod monitors and rules,
+the Prometheus operator doesn't reconcile `AlertmanagerConfig` resources from
+namespaces with the `openshift.io/user-monitoring: "false"` label. It allows
+application owners to opt out completely from UWM in case they deploy and run
+their own monitoring infrastructure (for instance with the [Monitoring Stack
+operator][monitoring-stack-operator]).
+
+In addition, the cluster admins can exclude specific user namespace(s) from UWM
+with the new `excludeUserNamespaces` field.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |-
+    enableUserWorkload: true
+    alertmanagerMain:
+      enableUserAlertmanagerConfig: true
+    excludeUserNamespaces: [foo,bar]
+```
+
+##### Dedicated UWM Alertmanager
+
+In some environments where cluster admins and UWM admins are different personas
+(e.g. OSD), it might not be acceptable for cluster admins to let users
+configure the Platform Alertmanager with `AlertmanagerConfig` resources because:
+* User configurations may break the Alertmanager configuration.
+* Processing of user alerts may slow down the alert notification pipeline.
+* Cluster admins don't want to deal with delivery errors for user notifications.
+
+At the same time, application owners want to configure their alert
+notifications without requesting external intervention.
+
+In this case, UWM admins have the possibility to deploy a dedicated
+Alertmanager. The configuration options are equivalent to the options
+exposed for the Platform Alertmanager and live under the `alertmanager` key
+in the UWM configmap.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: user-workload-monitoring-config
+  namespace: openshift-user-workload-monitoring
+data:
+  config.yaml: |-
+    alertmanager:
+      enabled: true
+      enableUserAlertmanagerConfig: true
+      logLevel: info
+      nodeSelector: {...}
```

Comment:
@simonpasquier will this be workers by default?

Reply:
Yes, it should be similar to what happens for the other monitoring components. When nodeSelector and tolerations aren't defined, all worker nodes are eligible and master nodes are excluded (because they are tainted).

Reply:
Excellent, the current prom-operator from UWM schedules itself on masters :)

```
$ oc -n openshift-user-workload-monitoring get pods prometheus-operator-68fd48b89d-6tp7s -o json | jq '.spec.tolerations[]'
{
  "effect": "NoSchedule",
  "key": "node-role.kubernetes.io/master",
  "operator": "Exists"
}
{
  "effect": "NoExecute",
  "key": "node.kubernetes.io/not-ready",
  "operator": "Exists",
  "tolerationSeconds": 300
}
{
  "effect": "NoExecute",
  "key": "node.kubernetes.io/unreachable",
  "operator": "Exists",
  "tolerationSeconds": 300
}
{
  "effect": "NoSchedule",
  "key": "node.kubernetes.io/memory-pressure",
  "operator": "Exists"
}
```

Reply:
yes this is following the OpenShift convention for operators -> https://github.com/openshift/enhancements/blob/master/CONVENTIONS.md#taints-and-tolerations